evalai/teams.py

```diff
@@ -2,8 +2,66 @@
 
 from click import echo
 
+from evalai.utils.teams import (
+                                get_participant_teams,
+                                create_participant_team,
+                                challenge_participate
+                               )
 
-@click.command()
-def teams():
-    """Example script."""
-    echo('Hello Teams!')
+
+@click.group(invoke_without_command=True)
+@click.pass_context
+def teams(ctx):
+    """
+    Teams and related options.
+    """
+    if ctx.invoked_subcommand is None:
+        welcome_text = ("Welcome to the EvalAI CLI. Use evalai"
+                        " teams --help for viewing all the options.")
+        echo(welcome_text)
+
+
+@click.command(name='list')
+def list_teams():
+    """
+    Used to list all the participant teams of a user.
+    Invoked by running `evalai teams list`
+    """
+    get_participant_teams()
+
+
+@click.command(name='create')
+def create_teams():
+    """
+    Used to create participant team.
+    Invoked by running `evalai teams create`
+    """
+    create_participant_team()
+
+
+@click.command(name='participate')
+@click.option('-c', '--challenge-id', type=int,
+              help="Challenge ID for participating.")
+@click.option('-pt', '--participant-team', type=int,
+              help="Challenge ID for participating.")
+def participate(challenge_id, participant_team):
+    """
+    Used to create participant team.
+    Invoked by running `evalai teams participate`
+    """
+    if challenge_id is None or participant_team is None:
```

Comment:
`Arguments only support a subset of the features of options due to their syntactical nature.` This is why I used options. I passed the required=True to overcome that. That's enough right?

Reply:
What features does options provide that will help you in this case that is missing in arguments?

Reply:
I don't see how it would be confusing. If we're consistent in the order and always have challenge ID followed by participant team, and this shows up in the help as well, I don't get why it would get confusing.
Rather it would be easier for the user as they would not need to type in more characters.

I'm going to spend some time checking this out myself and seeing what is the best way to structure this. We may have to rethink the design of the CLI tool if this pattern seems confusing or counter-intuitive.